### PR TITLE
Dev v2.6 remove flannel resource limits

### PR DESCRIPTION
**Pull request by vardhaman22**
related issue: https://github.com/rancher/rancher/issues/40228

### CODE REVIEW

**Review by kinarashah — CHANGES_REQUESTED** [review]
Build is failing, it's complaining about the missing `v1.22.17` which I suspect is because it was released in `release-v2.6` yesterday but the branch doesn't have those changes yet....Could you double check that both `dev-v2.6-0123-patches` and your local branch have the latest k8s versions?

File: rke/templates/flannel_v0.14.0-rancher2.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
+package templates
+
+const FlannelTemplateV0_14_0Rancher2 = `
```
Comment (kinarashah) [inline rke/templates/flannel_v0.14.0-rancher2.go:10]: Could you add a comment here for Rancher changelog listing the change so we can track the reasoning for a new template? example here: https://github.com/rancher/kontainer-driver-metadata/blob/dev-v2.6/rke/templates/nginx-v1.2.0-rancher2.go#L4

Comment (vardhaman22) [reply rke/templates/flannel_v0.14.0-rancher2.go:10]: thanks for pointing that out. 
added it.

File: rke/templates/flannel_v0.14.0-rancher2.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
+package templates
+
+/*
+FlannelTemplateV0_14_0Rancher2 is based on upstream flannel v0.14.0
+Source: https://raw.githubusercontent.com/flannel-io/flannel/v0.14.0/Documentation/kube-flannel.yml
+
+Rancher Changelog:
+- Remove resource limits for kube-flannel DaemonSet
```
Comment (snasovich) [inline rke/templates/flannel_v0.14.0-rancher2.go]: Let's reference upstream PR here so it doesn't look like Rancher-specific change - https://github.com/flannel-io/flannel/pull/1694

**Review by snasovich — DISMISSED** [review]
One minor comment, otherwise LGTM 👍

File: rke/templates/flannel_v0.14.0-rancher2.go (hunk 2 above)
Comment (vardhaman22) [reply rke/templates/flannel_v0.14.0-rancher2.go]: done

**Review by kinarashah — APPROVED** [review]

**Pull request merged**